core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -328,13 +330,23 @@ public static void validate() {
   }
 
   /**
-   * Gets the raw (no alias lookup) display configuration of a given scope.
-   *
-   * @param scope the property key scope
-   * @return a list of raw configurations inside the property scope
+   * @return a map from all configuration property names to their values; values may potentially be
+   *         null
+   */
+  public static Map<String, String> toMap() {
+    return toMap(ConfigurationValueOptions.defaults());
+  }
+
+  /**
+   * @param opts options for formatting the configuration values
+   * @return a map from all configuration property names to their values; values may potentially be
+   *         null
    */
-  public static List<ConfigProperty> getConfiguration(Scope scope) {
-    return CONF.getConfiguration(scope);
+  public static Map<String, String> toMap(ConfigurationValueOptions opts) {
+    Map<String, String> map = new HashMap<>();
+    // Cannot use Collectors.toMap because we support null keys.
+    CONF.keySet().forEach(key -> map.put(key.getName(), CONF.getOrDefault(key, null, opts)));
```

Comment:
like commented above, if we want `toMap` for instanced conf, it maybe worth to do delegation here too.

Reply:
done